src/lib/mina_lib/mina_lib.ml

```diff
@@ -1434,13 +1456,35 @@ let create ?wallets (config : Config.t) =
                       ~metadata:[ ("exn", Error_json.error_to_yojson err) ] ) )
               (fun () ->
                 O1trace.thread "manage_verifier_subprocess" (fun () ->
-                    Verifier.create ~logger:config.logger
-                      ~enable_internal_tracing:(Internal_tracing.is_enabled ())
-                      ~internal_trace_filename:"verifier-internal-trace.jsonl"
-                      ~proof_level:config.precomputed_values.proof_level
-                      ~constraint_constants:
-                        config.precomputed_values.constraint_constants
-                      ~pids:config.pids ~conf_dir:(Some config.conf_dir) () ) )
+                    let%bind verifier =
+                      Verifier.create ~logger:config.logger
+                        ~enable_internal_tracing:
+                          (Internal_tracing.is_enabled ())
+                        ~internal_trace_filename:"verifier-internal-trace.jsonl"
+                        ~proof_level:config.precomputed_values.proof_level
+                        ~constraint_constants:
+                          config.precomputed_values.constraint_constants
+                        ~pids:config.pids ~conf_dir:(Some config.conf_dir) ()
+                    in
+                    let%map () =
+                      if Mina_compile_config.itn_features then
+                        let ({ client_port; _ } : Node_addrs_and_ports.t) =
+                          config.gossip_net_params.addrs_and_ports
+                        in
+                        match%map
+                          Verifier.set_itn_logger_data verifier
+                            ~daemon_port:client_port
+                        with
+                        | Ok () ->
+                            ()
+                        | Error err ->
+                            [%log' warn config.logger]
+                              "Error when setting ITN logger data for \
+                               verifier: %s"
+                              (Error.to_string_hum err)
+                      else Deferred.unit
+                    in
+                    verifier ) )
```

Comment:
This code appears twice and the appearences seem to only differ by the module (`Prover` vs. `Verifier`), both of which seem to conveniently have the same signature to the extent required here. Let's extract this patch and make it a function!